src/provider/form-factory-provider.tsx

```diff
@@ -83,7 +83,7 @@ export const FormFactoryProvider: React.FC<FormFactoryProviderProps> = ({
   const registerForm = useCallback((formId: string, context: FormContextProps) => {
     if (!rootForm.current) {
       rootForm.current = context;
-    } else {
+    } else if (rootForm.current.formJson.name !== formId) {
```

Comment:
Doesn't `formId` have the uuid of the form? If that's the case then this might be wrong because the comparison seems to be with the form `name`

Reply:
It's currently based on [the form name](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/main/src/components/renderer/form/form-renderer.component.tsx#L67) (this was for troubleshooting purposes)